diff -ruN boswars-2.5-src-old/engine/pathfinder/astar.cpp boswars-2.5-src/engine/pathfinder/astar.cpp
--- boswars-2.5-src-old/engine/pathfinder/astar.cpp	2008-05-21 23:25:41.000000000 +0200
+++ boswars-2.5-src/engine/pathfinder/astar.cpp	2008-05-21 23:26:26.000000000 +0200
@@ -33,6 +33,8 @@
 --  Includes
 ----------------------------------------------------------------------------*/
 
+#include <stdlib.h>
+#include <stdio.h>
 #include "stratagus.h"
 #include "pathfinder.h"

diff -ruN boswars-2.5-src-old/engine/stratagus/stratagus.cpp boswars-2.5-src/engine/stratagus/stratagus.cpp
--- boswars-2.5-src-old/engine/stratagus/stratagus.cpp	2008-05-21 23:25:41.000000000 +0200
+++ boswars-2.5-src/engine/stratagus/stratagus.cpp	2008-05-21 23:26:54.000000000 +0200
@@ -160,6 +160,8 @@
 #include <time.h>
 #include <ctype.h>
 #include <sstream>
+#include <algorithm>
+#include <stdio.h>

 #ifndef _MSC_VER
 uinclude <unistd.h>

diff -ruN boswars-2.5-src-old/engine/pathfinder/astar.cpp boswars-2.5-src/engine/stratagus/translate.cpp
--- srcBU/boswars-2.5-src/engine/stratagus/translate.cpp	2008-01-26 16:02:54.000000000 -0700
+++ boswars-2.5-src/engine/stratagus/translate.cpp	2010-02-11 00:48:22.000000000 -0700
@@ -31,7 +31,8 @@
 /*----------------------------------------------------------------------------
 --  Includes
 ----------------------------------------------------------------------------*/
-
+#include <stdlib.h>
+#include <stdio.h>
 #include "stratagus.h"
 
 #include <map>

