diff -urN blobby-alpha-6/src/NetworkState.cpp fixed_copy/src/NetworkState.cpp
--- blobby-alpha-6/src/NetworkState.cpp	2006-10-29 12:31:42.000000000 +0000
+++ fixed_copy/src/NetworkState.cpp	2008-06-22 02:15:58.000000000 +0000
@@ -12,6 +12,7 @@
 #include "SoundManager.h"
 #include "LocalInputSource.h"
 #include "raknet/RakServer.h"
+#include <algorithm>
 
 NetworkSearchState::NetworkSearchState()
 {
diff -urN blobby-alpha-6/src/State.cpp fixed_copy/src/State.cpp
--- blobby-alpha-6/src/State.cpp	2006-10-29 12:36:56.000000000 +0000
+++ fixed_copy/src/State.cpp	2008-06-22 02:16:54.000000000 +0000
@@ -11,6 +11,7 @@
 
 #include <physfs.h>
 #include <sstream>
+#include <algorithm>
 
 State::State()
 {
