--- BallView.h   2003-11-19 17:49:31.000000000 +0100
+++ BallView.h~  2004-05-05 12:04:46.000000000 +0200
@@ -32,7 +32,7 @@
 
   static GLuint m_number[10];
 private:
-  void BallView::DrawTargetCircle();
+  void DrawTargetCircle();
   GLUquadricObj *m_quad;
 
   void DrawBall();

--- loadparts.cpp   2003-11-19 17:49:31.000000000 +0100
+++ loadparts.cpp~  2004-05-05 12:04:46.000000000 +0200
@@ -245,7 +245,11 @@
 
 	while ('\\' == line[l-1]) {
             // concat next line(s)
+#if (defined __ia64__) || (defined __alpha__) || (defined __powerpc64__) || (defined __s390__) || (defined __x86_64__)
+	    int bufsize = clamp((u_long) 0, sizeof(line)-l, sizeof(line)-1);
+#else
 	    int bufsize = clamp(0U, sizeof(line)-l, sizeof(line)-1);
+#endif
 	    fgets(&line[l-2], bufsize, fp);
 	    if (feof((FILE*)fp)) break;
 	    l = strlen(line);
								     
--- matrix   2002-03-06 17:46:01.000000000 +0100
+++ matrix~  2005-01-06 19:43:32.037852232 +0100
@@ -42,15 +42,6 @@
 
 /* __BEGIN__BEGIN__ */
 //__NAMESPACE_BEGIN(ese);
-#if (!defined __GNUG__) || (__GNUC__ == 3)
-template<typename T>
-void swap(T& a, T& b)
-{
-    T c = a;
-    a = b;
-    b = c;
-}
-#endif
 
 // matrix_size MUST BE larger than 1
 template<int matrix_size, typename _float_t = float>
@@ -174,9 +165,10 @@
 		    mx = x;
 		}
 	    }
-	    swap(pivot[k], pivot[piv]);
 
 	    int kk = pivot[k];
+            pivot[k] = pivot[piv];
+            pivot[piv] = kk;
 	    float_t t = m[kk][kk];
 	    if (0 == t) {
 		return 0;
@@ -222,9 +214,10 @@
 		    mx = x;
 		}
 	    }
-	    swap(pivot[k], pivot[piv]);
 
 	    int kk = pivot[kk];
+	    pivot[k] = pivot[piv];
+	    pivot[piv] = kk;
 	    double t = a[kk][kk];
 	    if (0 == t) {
 		return 0;
@@ -283,7 +276,8 @@
 	    }
 	    ik = ip[j];
 	    if (j != k) {
-		swap(ip[j], ip[k]);
+		ip[j] = ip[k];
+		ip[k] = ik;
 		det = -det;
 	    }
 	    u = a[ik][k];
 	    
--- configure   2003-08-16 07:54:31.000000000 +0200
+++ configure~  2006-07-24 14:04:49.000000000 +0200
@@ -5515,7 +5515,7 @@
 fi
 
 
-CFLAGS="-I $x_includes"
+CFLAGS="-I/ $x_includes"
 LDFLAGS="-L/usr/local/lib -L$x_libraries -L$libdir"
 CXXFLAGS="$CFLAGS -Wall"
