--- spring_0.76b1/build/intopts.py.orig	2008-06-20 14:52:03.000000000 +0200
+++ spring_0.76b1/build/intopts.py	2008-06-20 14:50:51.000000000 +0200
@@ -1,8 +1,8 @@
+import os
 LINKFLAGS = ['-Wl,-E', '-Wl,-rpath,/usr/lib']
 LIBPATH = ['rts/lib/streflop', '/usr/lib']
 LIBS = ['freetype', 'z', 'SDL', 'pthread', 'openal', 'boost_regex-mt', 'boost_thread-mt', 'GL', 'GLU', 'GLEW', 'z', 'freetype', 'IL', 'ILU', 'openal', 'vorbisfile', 'vorbis', 'ogg', 'SDL', 'python2.5', 'streflop']
-CCFLAGS = ['-fsingle-precision-constant', '-frounding-math', '-fsignaling-nans', '-mieee-fp', '-O2', '-pipe', '-fno-strict-aliasing', '-fvisibility=hidden', '-mfpmath=387', '-pthread']
-CXXFLAGS = ['-fsingle-precision-constant', '-frounding-math', '-fsignaling-nans', '-mieee-fp', '-O2', '-pipe', '-fno-strict-aliasing', '-fvisibility=hidden', '-mfpmath=387', '-pthread']
+CXXFLAGS = [os.getenv('CXXFLAGS').split(), '-fsingle-precision-constant', '-frounding-math', '-fsignaling-nans', '-mieee-fp', '-fno-strict-aliasing', '-fvisibility=hidden', '-mfpmath=387', '-pthread']
 CPPDEFINES = ['_SZ_ONE_DIRECTORY', ['_GNU_SOURCE', '1'], '_REENTRANT']
 CPPPATH = ['rts', 'rts/System', 'rts/lib/luabind', 'rts/lib/lua/include', '/usr/include', '/usr/include/GL', '/usr/include/freetype2', '/usr/include/SDL', '/usr/include/ogg', '/usr/include/vorbis', '/usr/include/python2.5', '/usr/include/python2.4', '/usr/include/java']
 CC = 'gcc'
