diff -ruN dwarftherapist-read-only/src/gamedatareader.cpp dwarftherapist-read-only-build/src/gamedatareader.cpp
--- dwarftherapist-read-only/src/gamedatareader.cpp	2009-11-08 14:55:52.000000000 -0500
+++ dwarftherapist-read-only-build/src/gamedatareader.cpp	2009-11-08 15:44:37.000000000 -0500
@@ -31,7 +31,7 @@
 	: QObject(parent)
 {
 	QDir working_dir = QDir::current();
-	QString filename = working_dir.absoluteFilePath("etc/game_data.ini");
+	QString filename = "/usr/share/dwarftherapist/game_data.ini";
 	m_data_settings = new QSettings(filename, QSettings::IniFormat);
 
 	m_data_settings->beginGroup("labors");
diff -ruN dwarftherapist-read-only/src/memorylayout.cpp dwarftherapist-read-only-build/src/memorylayout.cpp
--- dwarftherapist-read-only/src/memorylayout.cpp	2009-11-08 14:55:52.000000000 -0500
+++ dwarftherapist-read-only-build/src/memorylayout.cpp	2009-11-08 15:44:59.000000000 -0500
@@ -18,7 +18,7 @@
 #ifdef _OSX
 	QString subdir = "osx";
 #endif
-        QString filename = working_dir.absoluteFilePath("etc/memory_layouts/" + subdir + "/" + m_game_version + ".ini");
+        QString filename = "/usr/share/dwarftherapist/memory_layouts/" + subdir + "/" + m_game_version + ".ini";
 		m_data = new QSettings(filename, QSettings::IniFormat);
 		load_data();
 	}
