--- ./src/lisp/lexer.cpp	2007-09-14 08:43:03.000000000 +0300
+++ ./src/lisp/lexer.cpp~	2008-10-10 12:10:13.000000000 +0200
@@ -19,6 +19,7 @@
 //  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 #include <sstream>
 #include <stdexcept>
+#include <cstring>
 
 #include "lexer.hpp"
 #include "translation.hpp"

--- ./src/lisp/parser.cpp	2007-09-14 08:43:03.000000000 +0300
+++ ./src/lisp/parser.cpp~	2008-10-10 12:13:01.000000000 +0200
@@ -20,7 +20,7 @@
 #include <sstream>
 #include <stdexcept>
 #include <fstream>
-
+#include <cstring>
 #include <cassert>
 
 #include "parser.hpp"
 
--- ./src/ui/menusystem.cpp	2007-09-14 08:43:03.000000000 +0300
+++ ./src/ui/menusystem.cpp~	2008-10-10 12:15:30.000000000 +0200
@@ -1,7 +1,7 @@
 #include <iostream>
 
-#include "settings.h"
-#include "canvas.h"
+#include "../settings.h"
+#include "../canvas.h"
 
 #include "menusystem.h"
 #include "menu.h"
 
--- ./src/menumanager.h	2007-09-14 08:43:03.000000000 +0300
+++ ./src/menumanager.h~	2008-10-10 12:59:21.000000000 +0200
@@ -7,6 +7,8 @@
 #ifndef MENUMANAGER_H
 #define MENUMANAGER_H
 
+#include "ui/menuaction.h"
+
 class Main;
 
 class ControlOptions;
 
--- ./src/sdl_driver.cpp	2007-09-14 08:43:03.000000000 +0300
+++ ./src/sdl_driver.cpp~	2008-10-10 12:44:39.000000000 +0200
@@ -47,7 +47,7 @@
     Canvas::initScreen();
 
     const int NUM_STICKS = SDL_NumJoysticks();
-    sticks = (SDL_Joystick **) std::malloc(sizeof(SDL_Joystick *) * NUM_STICKS);
+    sticks = (SDL_Joystick **) malloc(sizeof(SDL_Joystick *) * NUM_STICKS);
     for (int i=0;i<NUM_STICKS;i++)
         sticks[i] = SDL_JoystickOpen(i);
 
@@ -84,7 +84,7 @@
 {
   SDL_ShowCursor(SDL_ENABLE);
 
-  std::free(sticks);
+  free(sticks);
 
   Canvas::shutdown();
 
--- ./src/videooptions.cpp	2007-09-14 08:43:03.000000000 +0300
+++ ./src/videooptions.cpp~	2008-10-10 12:46:10.000000000 +0200
@@ -1,5 +1,6 @@
 #include <vector>
 #include <sstream>
+#include <algorithm>
 
 #include <SDL/SDL.h>
 
