--- src/cal3d/coreskeleton.cpp	2008-10-26 19:33:21.000000000 +0200
+++ src/cal3d/coreskeleton.cpp~	2008-10-27 00:11:39.000000000 +0200
@@ -18,7 +18,7 @@
 #include "cal3d/corebone.h"
 #include "cal3d/coremodel.h"
 #include "cal3d/coresubmesh.h"
-
+#include <cstring>
 
 CalCoreSkeleton::CalCoreSkeleton()
 {
 
--- src/cal3d_converter.cpp	2008-10-26 19:33:22.000000000 +0200
+++ src/cal3d_converter.cpp~	2008-10-27 00:16:13.000000000 +0200
@@ -10,7 +10,7 @@
 //****************************************************************************//
 
 #include "cal3d/cal3d.h"
-
+#include <cstring>
 #define SKELETON 0
 #define MESH 1
 #define ANIMATION 2

--- src/cal3d/loader.cpp	2008-10-26 19:33:21.000000000 +0200
+++ src/cal3d/loader.cpp~	2008-10-27 00:25:30.000000000 +0200
@@ -33,7 +33,7 @@
 #include "cal3d/tinyxml.h"
 #include "cal3d/streamsource.h"
 #include "cal3d/buffersource.h"
-
+#include <memory>
 using namespace cal3d;
 
 int CalLoader::loadingMode;
