--- ../pkg/usr/bin/planeshift~	2008-03-16 02:36:37.000000000 +0100
+++ ../pkg/usr/bin/planeshift	2008-03-16 02:39:13.000000000 +0100
@@ -1,8 +1,6 @@
 #!/bin/bash
 
-R=$PWD/$0
-R=${R%/*}
-cd $R
+cd /opt/PlaneShift
 chmod 770 psclient.bin
 export LD_LIBRARY_PATH=libs/:\$LD_LIBRARY_PATH
 exec ./psclient.bin $@
--- ../pkg/usr/bin/planeshift-setup~	2008-03-16 02:36:37.000000000 +0100
+++ ../pkg/usr/bin/planeshift-setup	2008-03-16 02:40:09.000000000 +0100
@@ -1,8 +1,6 @@
 #!/bin/bash
 
-R=$PWD/$0
-R=${R%/*}
-cd $R
+cd /opt/PlaneShift
 chmod 770 pssetup.bin
 export LD_LIBRARY_PATH=libs/:\$LD_LIBRARY_PATH
 exec ./pssetup.bin $@
--- ../pkg/usr/bin/planeshift-updater~	2008-03-16 02:36:37.000000000 +0100
+++ ../pkg/usr/bin/planeshift-updater	2008-03-16 02:40:35.000000000 +0100
@@ -1,8 +1,6 @@
 #!/bin/bash
 
-R=$PWD/$0
-R=${R%/*}
-cd $R
+cd /opt/PlaneShift
 chmod 770 psupdater.bin
 export LD_LIBRARY_PATH=libs/:\$LD_LIBRARY_PATH
 exec ./psupdater.bin $@
