--- src/gengine/OptionAgent.cpp~	2006-02-12 12:59:06.000000000 +0100
+++ src/gengine/OptionAgent.cpp	2008-04-26 23:47:56.000000000 +0200
@@ -25,6 +25,7 @@
 
 #include <string.h> //strlen
 #include <locale.h> //setlocale
+#include <cstdlib> //getenv
 
 #ifndef LC_MESSAGES
 #define LC_MESSAGES LC_CTYPE

--- src/gengine/StringTool.cpp~	2007-06-30 16:55:12.000000000 +0200
+++ src/gengine/StringTool.cpp	2008-04-26 23:43:03.000000000 +0200
@@ -9,6 +9,7 @@
 #include "StringTool.h"
 
 #include <sstream>
+#include <stdlib.h>
 
 //-----------------------------------------------------------------
 /**

--- src/level/Landslip.cpp~	2004-08-27 13:37:56.000000000 +0200
+++ src/level/Landslip.cpp	2008-04-26 23:58:26.000000000 +0200
@@ -6,6 +6,7 @@
  * the Free Software Foundation; either version 2 of the License, or
  * (at your option) any later version.
  */
+#include <string.h>
 #include "Landslip.h"
 
 #include "Rules.h"
