--- src/tools/trackgen/track.orig.cpp	2005-08-24 14:19:48.000000000 -0500
+++ src/tools/trackgen/track.cpp	2005-10-04 02:15:55.000000000 -0500
@@ -392,7 +392,7 @@
 	    trackvertices[3*nbvert]   = x;			\
 	    trackvertices[3*nbvert+1] = y;			\
 	    trackvertices[3*nbvert+2] = z;			\
-	    trackindices[nbvert]      = nbvert++;		\
+	    trackindices[nbvert]      = nbvert+1;		\
 	}							\
     } while (0)
 
@@ -403,7 +403,7 @@
 	    trackvertices[3*nbvert]   = x;			\
 	    trackvertices[3*nbvert+1] = y;			\
 	    trackvertices[3*nbvert+2] = z;			\
-	    trackindices[nbvert]      = nbvert++;		\
+	    trackindices[nbvert]      = nbvert+1;		\
 	    printf("x=%f y=%f z=%f  u=%f v=%f\n",		\
 		   (tdble)(x), (tdble)(y), (tdble)(z),		\
 		   (tdble)(t1), (tdble)(t2));			\
